pcmpl-args.el

```diff
@@ -3446,58 +3449,62 @@ It is suffixed with a slash."
                        (expand-file-name "~/.password-store"))))
     (concat directory "/")))
 
-(defun pcmpl-args-pass-find (&rest find-args)
+(defun pcmpl-args-pass-find (&optional type)
   "Return a list of password-store entries.
 By default, return all directories and files in password-store.
-These can be limited by `FIND-ARGS'.
-
-`FIND-ARGS' is a list of find (GNU/findutils) arguments.  For
-example, to get only directories:
-
-\(pcmpl-args-pass-find \"-type\" \"d\")"
-  (let* ((prefix (pcmpl-args-pass-prefix))
-         (args `("find" "-L" ,prefix
-                 "(" "-name" ".git*" "-o" "-name" ".gpg-id" ")" "-prune"
-                 "-o" ,@find-args "-print"))
-         (rx (concat "^" (regexp-quote prefix) "\\(.+?\\)\\(?:\\.gpg\\)?$")))
-    (with-temp-buffer
-      (apply #'pcmpl-args-process-file args)
-      (goto-char (point-min))
-      (save-match-data
-        (while (search-forward-regexp rx nil t)
-          (replace-match "\\1")))
-      (beginning-of-line)
-      (let (lines)
-        (while (progn (push (string-trim-right (thing-at-point 'line t)) lines)
-                      (forward-line -1)
-                      (not (bobp))))
-        lines))))
+These can be limited by TYPE.
+
+If TYPE is :files, return only files.  If TYPE is :directories,
+return only directories."
+  (let ((dir (pcmpl-args-pass-prefix)))
+
+    (cl-labels
+        ((no-git (dir) (not (string-suffix-p "/.git" dir)))
+         (chop-dir (entry) (string-remove-prefix dir entry))
+         (chop-ext (entry) (string-remove-suffix ".gpg" entry))
+         (chop (entry) (chop-dir (chop-ext entry)))
+         (dotp (file) (string-prefix-p "." (file-name-base file)))
+         (gpgp (file) (string-suffix-p ".gpg" file))
+         (dot-or-gpg-p (file) (or (dotp file) (gpgp file))))
+
+      (cl-case type
+        (:files
+         (thread-last (directory-files-recursively dir "\\.gpg\\'" nil #'no-git t)
+           (mapcar #'chop)))
+        (:directories
+         (thread-last (directory-files-recursively dir ".*" t #'no-git t)
+           (cl-delete-if #'dot-or-gpg-p)
+           (mapcar #'chop-dir)))
+        (t
+         (thread-last (directory-files-recursively dir ".*" t #'no-git t)
+           (cl-delete-if #'dotp)
+           (mapcar #'chop)))))))
```

Comment:
What is the motivation for reimplementing without find?

Reply:
No external dependencies. It is easier to work with a list of strings, in comparison to a buffer. In fact, I found a bug, it was missing one pass entry.